invoices/settlement_interceptor.go

```diff
@@ -0,0 +1,224 @@
+package invoices
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// InterceptSession creates a session that is returned to the caller when an
+// invoice is submitted to this service. This session allows the caller to block
+// until the invoice is processed.
+type InterceptSession struct {
+	HtlcModifyRequest
```

Comment:
I find it structurally weird to embed a session with a request, can we do `Req HtlcModifyRequest` instead?